core/src/main/java/org/pentaho/di/core/util/StringUtil.java

```diff
@@ -613,4 +613,46 @@ public static String safeToLowerCase( final Object obj ) {
       return obj.toString().toLowerCase();
     }
   }
+
+  /**
+   * Removes all instances of the specified character from the start of the given {@link String}.
+   *
+   * @param s the {@link String} to trim
+   * @param c the character to remove from the {@link String}
+   * @return a new string with all instances of the specified character removed from the start
+   */
+  public static String trimStart( final String s, char c ) {
+    if ( isEmpty( s ) ) {
+      return s;
+    }
+
+    String t = s;
+
+    while ( t.charAt( 0 ) == c ) {
+      t = t.substring( 1 );
+    }
+
+    return  t;
+  }
+
+  /**
+   * Removes all instances of the specified character from the end of the given {@link String}.
+   *
+   * @param s the {@link String} to trim
+   * @param c the character to remove from the {@link String}
+   * @return a new string with all instances of the specified character removed from the end
+   */
+  public static String trimEnd( final String s, char c ) {
```

Comment:
Same criticism as above. Something more like this performs much better and doesn't create tons of temporary strings.

  public static String rtrim( String source, char c ) {
    if ( source == null ) {
      return null;
    }
    int max = source.length();
    while ( max > 0 && ( source.charAt( max - 1 ) == c ) ) {
      max--;
    }
    return source.substring( 0, max );
  }